// @orrery/core/dist/chunk-EXYPMSUR.js
// src/constants.ts
var SKY = "\u7532\u4E59\u4E19\u4E01\u620A\u5DF1\u5E9A\u8F9B\u58EC\u7678";
var EARTH = "\u5B50\u4E11\u5BC5\u536F\u8FB0\u5DF3\u5348\u672A\u7533\u9149\u620C\u4EA5";
var SKY_KR = "\uAC11\uC744\uBCD1\uC815\uBB34\uAE30\uACBD\uC2E0\uC784\uACC4";
var EARTH_KR = "\uC790\uCD95\uC778\uBB18\uC9C4\uC0AC\uC624\uBBF8\uC2E0\uC720\uC220\uD574";
var YANGGAN = ["\u7532", "\u4E19", "\u620A", "\u5E9A", "\u58EC"];
var RELATIONS = [
  { hanja: "\u6BD4\u80A9", hangul: "\uBE44\uACAC" },
  { hanja: "\u52AB\u8CA1", hangul: "\uAC81\uC7AC" },
  { hanja: "\u98DF\u795E", hangul: "\uC2DD\uC2E0" },
  { hanja: "\u50B7\u5B98", hangul: "\uC0C1\uAD00" },
  { hanja: "\u504F\u8CA1", hangul: "\uD3B8\uC7AC" },
  { hanja: "\u6B63\u8CA1", hangul: "\uC815\uC7AC" },
  { hanja: "\u504F\u5B98", hangul: "\uD3B8\uAD00" },
  { hanja: "\u6B63\u5B98", hangul: "\uC815\uAD00" },
  { hanja: "\u504F\u5370", hangul: "\uD3B8\uC778" },
  { hanja: "\u6B63\u5370", hangul: "\uC815\uC778" }
];
var METEORS_12 = [
  { hanja: "\u9577\u751F", hangul: "\uC7A5\uC0DD" },
  { hanja: "\u6C90\u6D74", hangul: "\uBAA9\uC695" },
  { hanja: "\u51A0\u5E36", hangul: "\uAD00\uB300" },
  { hanja: "\u4E7E\u797F", hangul: "\uAC74\uB85D" },
  { hanja: "\u5E1D\u65FA", hangul: "\uC81C\uC655" },
  { hanja: "\u8870", hangul: "\uC1E0" },
  { hanja: "\u75C5", hangul: "\uBCD1" },
  { hanja: "\u6B7B", hangul: "\uC0AC" },
  { hanja: "\u5893", hangul: "\uBB18" },
  { hanja: "\u7D76", hangul: "\uC808" },
  { hanja: "\u80CE", hangul: "\uD0DC" },
  { hanja: "\u990A", hangul: "\uC591" }
];
var SPIRITS_12 = [
  { hanja: "\u52AB\u6BBA", hangul: "\uAC81\uC0B4" },
  { hanja: "\u707D\u6BBA", hangul: "\uC7AC\uC0B4" },
  { hanja: "\u5929\u6BBA", hangul: "\uCC9C\uC0B4" },
  { hanja: "\u5730\u6BBA", hangul: "\uC9C0\uC0B4" },
  { hanja: "\u5E74\u6BBA", hangul: "\uC5F0\uC0B4" },
  { hanja: "\u6708\u6BBA", hangul: "\uC6D4\uC0B4" },
  { hanja: "\u4EA1\u8EAB", hangul: "\uB9DD\uC2E0" },
  { hanja: "\u5C07\u661F", hangul: "\uC7A5\uC131" },
  { hanja: "\u6500\u978D", hangul: "\uBC18\uC548" },
  { hanja: "\u9A5B\u99AC", hangul: "\uC5ED\uB9C8" },
  { hanja: "\u516D\u5BB3", hangul: "\uC721\uD574" },
  { hanja: "\u83EF\u84CB", hangul: "\uD654\uAC1C" }
];
var BRANCH_ELEMENT = {
  "\u5BC5": "tree",
  "\u536F": "tree",
  "\u5DF3": "fire",
  "\u5348": "fire",
  "\u8FB0": "earth",
  "\u672A": "earth",
  "\u4E11": "earth",
  "\u620C": "earth",
  "\u7533": "metal",
  "\u9149": "metal",
  "\u5B50": "water",
  "\u4EA5": "water"
};
var STEM_INFO = {
  "\u7532": { yinyang: "+", element: "tree" },
  "\u4E59": { yinyang: "-", element: "tree" },
  "\u4E19": { yinyang: "+", element: "fire" },
  "\u4E01": { yinyang: "-", element: "fire" },
  "\u620A": { yinyang: "+", element: "earth" },
  "\u5DF1": { yinyang: "-", element: "earth" },
  "\u5E9A": { yinyang: "+", element: "metal" },
  "\u8F9B": { yinyang: "-", element: "metal" },
  "\u58EC": { yinyang: "+", element: "water" },
  "\u7678": { yinyang: "-", element: "water" }
};
var ELEMENT_HANJA = {
  tree: "\u6728",
  fire: "\u706B",
  earth: "\u571F",
  metal: "\u91D1",
  water: "\u6C34"
};
var TRIPLE_COMPOSES = [
  ["\u5BC5", "\u5348", "\u620C"],
  // 火局
  ["\u5DF3", "\u9149", "\u4E11"],
  // 金局
  ["\u7533", "\u5B50", "\u8FB0"],
  // 水局
  ["\u4EA5", "\u536F", "\u672A"]
  // 木局
];
var TRIPLE_COMPOSE_ELEMENTS = {
  "\u5BC5,\u5348,\u620C": "fire",
  "\u5DF3,\u9149,\u4E11": "metal",
  "\u7533,\u5B50,\u8FB0": "water",
  "\u4EA5,\u536F,\u672A": "tree"
};
var HALF_COMPOSES = {
  "\u5BC5,\u5348": ["\u534A\u5408", "fire"],
  "\u5348,\u620C": ["\u534A\u5408", "fire"],
  "\u5DF3,\u9149": ["\u534A\u5408", "metal"],
  "\u9149,\u4E11": ["\u534A\u5408", "metal"],
  "\u7533,\u5B50": ["\u534A\u5408", "water"],
  "\u5B50,\u8FB0": ["\u534A\u5408", "water"],
  "\u4EA5,\u536F": ["\u534A\u5408", "tree"],
  "\u536F,\u672A": ["\u534A\u5408", "tree"]
};
var DIRECTIONAL_COMPOSES = [
  ["\u5BC5", "\u536F", "\u8FB0"],
  // 東方木
  ["\u5DF3", "\u5348", "\u672A"],
  // 南方火
  ["\u7533", "\u9149", "\u620C"],
  // 西方金
  ["\u4EA5", "\u5B50", "\u4E11"]
  // 北方水
];
var DIRECTIONAL_COMPOSE_ELEMENTS = {
  "\u5BC5,\u536F,\u8FB0": "tree",
  "\u5DF3,\u5348,\u672A": "fire",
  "\u7533,\u9149,\u620C": "metal",
  "\u4EA5,\u5B50,\u4E11": "water"
};
var STEM_COMBINES = {
  "\u7532,\u5DF1": ["\u5408", "earth"],
  "\u4E59,\u5E9A": ["\u5408", "metal"],
  "\u4E19,\u8F9B": ["\u5408", "water"],
  "\u4E01,\u58EC": ["\u5408", "tree"],
  "\u620A,\u7678": ["\u5408", "fire"]
};
var STEM_CLASHES = {
  "\u7532,\u5E9A": "\u6C96",
  "\u4E59,\u8F9B": "\u6C96",
  "\u4E19,\u58EC": "\u6C96",
  "\u4E01,\u7678": "\u6C96"
};
var BRANCH_COMBINES_6 = {
  "\u5B50,\u4E11": ["\u5408", "earth"],
  "\u5BC5,\u4EA5": ["\u5408", "tree"],
  "\u536F,\u620C": ["\u5408", "fire"],
  "\u8FB0,\u9149": ["\u5408", "metal"],
  "\u5DF3,\u7533": ["\u5408", "water"],
  "\u5348,\u672A": ["\u5408", "fire"]
};
var BRANCH_CLASHES = {
  "\u5B50,\u5348": "\u6C96",
  "\u4E11,\u672A": "\u6C96",
  "\u5BC5,\u7533": "\u6C96",
  "\u536F,\u9149": "\u6C96",
  "\u8FB0,\u620C": "\u6C96",
  "\u5DF3,\u4EA5": "\u6C96"
};
var BRANCH_BREAKS = {
  "\u5B50,\u9149": "\u7834",
  "\u4E11,\u8FB0": "\u7834",
  "\u5BC5,\u4EA5": "\u7834",
  "\u536F,\u5348": "\u7834",
  "\u5DF3,\u7533": "\u7834",
  "\u672A,\u620C": "\u7834"
};
var BRANCH_HARMS = {
  "\u5B50,\u672A": "\u5BB3",
  "\u4E11,\u5348": "\u5BB3",
  "\u5BC5,\u5DF3": "\u5BB3",
  "\u536F,\u8FB0": "\u5BB3",
  "\u7533,\u4EA5": "\u5BB3",
  "\u9149,\u620C": "\u5BB3"
};
var BRANCH_PUNISHMENTS = {
  "\u5BC5,\u5DF3": ["\u5211", "\u7121\u6069"],
  "\u5DF3,\u7533": ["\u5211", "\u7121\u6069"],
  "\u7533,\u5BC5": ["\u5211", "\u7121\u6069"],
  "\u4E11,\u620C": ["\u5211", "\u7121\u79AE"],
  "\u620C,\u672A": ["\u5211", "\u7121\u79AE"],
  "\u672A,\u4E11": ["\u5211", "\u7121\u79AE"],
  "\u5B50,\u536F": ["\u5211", "\u76F8\u5211"],
  "\u536F,\u5B50": ["\u5211", "\u76F8\u5211"]
};
var BRANCH_SELF_PUNISHMENTS = /* @__PURE__ */ new Set(["\u8FB0", "\u5348", "\u9149", "\u4EA5"]);
var BRANCH_WONJIN = {
  "\u5B50,\u672A": "\u6028\u55D4",
  "\u4E11,\u5348": "\u6028\u55D4",
  "\u5BC5,\u9149": "\u6028\u55D4",
  "\u536F,\u7533": "\u6028\u55D4",
  "\u8FB0,\u4EA5": "\u6028\u55D4",
  "\u5DF3,\u620C": "\u6028\u55D4"
};
var BRANCH_GWIMUN = {
  "\u5B50,\u9149": "\u9B3C\u9580",
  "\u4E11,\u5348": "\u9B3C\u9580",
  "\u5BC5,\u672A": "\u9B3C\u9580",
  "\u536F,\u7533": "\u9B3C\u9580",
  "\u8FB0,\u4EA5": "\u9B3C\u9580",
  "\u5DF3,\u620C": "\u9B3C\u9580"
};
var YANGIN_MAP = {
  "\u7532": "\u536F",
  "\u4E19": "\u5348",
  "\u620A": "\u5348",
  "\u5E9A": "\u9149",
  "\u58EC": "\u5B50"
};
var BAEKHO_PILLARS = /* @__PURE__ */ new Set(["\u7532\u8FB0", "\u4E59\u672A", "\u4E19\u620C", "\u4E01\u4E11", "\u620A\u8FB0", "\u58EC\u620C", "\u7678\u4E11"]);
var GOEGANG_PILLARS = /* @__PURE__ */ new Set(["\u5E9A\u8FB0", "\u5E9A\u620C", "\u58EC\u8FB0", "\u620A\u620C"]);
var DOHWA_MAP = {
  "\u5BC5": "\u536F",
  "\u5348": "\u536F",
  "\u620C": "\u536F",
  "\u7533": "\u9149",
  "\u5B50": "\u9149",
  "\u8FB0": "\u9149",
  "\u5DF3": "\u5348",
  "\u9149": "\u5348",
  "\u4E11": "\u5348",
  "\u4EA5": "\u5B50",
  "\u536F": "\u5B50",
  "\u672A": "\u5B50"
};
var CHEONUL_MAP = {
  "\u7532": ["\u4E11", "\u672A"],
  "\u620A": ["\u4E11", "\u672A"],
  "\u5E9A": ["\u4E11", "\u672A"],
  "\u4E59": ["\u5B50", "\u7533"],
  "\u5DF1": ["\u5B50", "\u7533"],
  "\u4E19": ["\u4EA5", "\u9149"],
  "\u4E01": ["\u4EA5", "\u9149"],
  "\u8F9B": ["\u5348", "\u5BC5"],
  "\u58EC": ["\u5DF3", "\u536F"],
  "\u7678": ["\u5DF3", "\u536F"]
};
var CHEONDUK_MAP = {
  "\u5BC5": "\u4E01",
  "\u536F": "\u7533",
  "\u8FB0": "\u58EC",
  "\u5DF3": "\u8F9B",
  "\u5348": "\u4EA5",
  "\u672A": "\u7532",
  "\u7533": "\u7678",
  "\u9149": "\u5BC5",
  "\u620C": "\u4E19",
  "\u4EA5": "\u4E59",
  "\u5B50": "\u5DF3",
  "\u4E11": "\u5E9A"
};
var WOLDUK_MAP = {
  "\u5BC5": "\u4E19",
  "\u5348": "\u4E19",
  "\u620C": "\u4E19",
  "\u7533": "\u58EC",
  "\u5B50": "\u58EC",
  "\u8FB0": "\u58EC",
  "\u5DF3": "\u5E9A",
  "\u9149": "\u5E9A",
  "\u4E11": "\u5E9A",
  "\u4EA5": "\u7532",
  "\u536F": "\u7532",
  "\u672A": "\u7532"
};
var MUNCHANG_MAP = {
  "\u7532": "\u5DF3",
  "\u4E59": "\u5348",
  "\u4E19": "\u7533",
  "\u4E01": "\u9149",
  "\u620A": "\u7533",
  "\u5DF1": "\u9149",
  "\u5E9A": "\u4EA5",
  "\u8F9B": "\u5B50",
  "\u58EC": "\u5BC5",
  "\u7678": "\u536F"
};
var HONGYEOM_PILLARS = /* @__PURE__ */ new Set(["\u7532\u5348", "\u4E19\u5BC5", "\u4E01\u672A", "\u620A\u8FB0", "\u5E9A\u620C", "\u8F9B\u9149", "\u58EC\u5B50"]);
var GEUMYEO_MAP = {
  "\u7532": "\u8FB0",
  "\u4E59": "\u5DF3",
  "\u4E19": "\u672A",
  "\u4E01": "\u7533",
  "\u620A": "\u672A",
  "\u5DF1": "\u7533",
  "\u5E9A": "\u620C",
  "\u8F9B": "\u4EA5",
  "\u58EC": "\u4E11",
  "\u7678": "\u5BC5"
};
var METEOR_LOOKUP = {
  "\uAC11\uD574": 0,
  "\uC744\uC624": 0,
  "\uBCD1\uC778": 0,
  "\uC815\uC720": 0,
  "\uBB34\uC778": 0,
  "\uAE30\uC720": 0,
  "\uACBD\uC0AC": 0,
  "\uC2E0\uC790": 0,
  "\uC784\uC2E0": 0,
  "\uACC4\uBB18": 0,
  "\uAC11\uC790": 1,
  "\uC744\uC0AC": 1,
  "\uBCD1\uBB18": 1,
  "\uC815\uC2E0": 1,
  "\uBB34\uBB18": 1,
  "\uAE30\uC2E0": 1,
  "\uACBD\uC624": 1,
  "\uC2E0\uD574": 1,
  "\uC784\uC720": 1,
  "\uACC4\uC778": 1,
  "\uAC11\uCD95": 2,
  "\uC744\uC9C4": 2,
  "\uBCD1\uC9C4": 2,
  "\uC815\uBBF8": 2,
  "\uBB34\uC9C4": 2,
  "\uAE30\uBBF8": 2,
  "\uACBD\uBBF8": 2,
  "\uC2E0\uC220": 2,
  "\uC784\uC220": 2,
  "\uACC4\uCD95": 2,
  "\uAC11\uC778": 3,
  "\uC744\uBB18": 3,
  "\uBCD1\uC0AC": 3,
  "\uC815\uC624": 3,
  "\uBB34\uC0AC": 3,
  "\uAE30\uC624": 3,
  "\uACBD\uC2E0": 3,
  "\uC2E0\uC720": 3,
  "\uC784\uD574": 3,
  "\uACC4\uC790": 3,
  "\uAC11\uBB18": 4,
  "\uC744\uC778": 4,
  "\uBCD1\uC624": 4,
  "\uC815\uC0AC": 4,
  "\uBB34\uC624": 4,
  "\uAE30\uC0AC": 4,
  "\uACBD\uC720": 4,
  "\uC2E0\uC2E0": 4,
  "\uC784\uC790": 4,
  "\uACC4\uD574": 4,
  "\uAC11\uC9C4": 5,
  "\uC744\uCD95": 5,
  "\uBCD1\uBBF8": 5,
  "\uC815\uC9C4": 5,
  "\uBB34\uBBF8": 5,
  "\uAE30\uC9C4": 5,
  "\uACBD\uC220": 5,
  "\uC2E0\uBBF8": 5,
  "\uC784\uCD95": 5,
  "\uACC4\uC220": 5,
  "\uAC11\uC0AC": 6,
  "\uC744\uC790": 6,
  "\uBCD1\uC2E0": 6,
  "\uC815\uBB18": 6,
  "\uBB34\uC2E0": 6,
  "\uAE30\uBB18": 6,
  "\uACBD\uD574": 6,
  "\uC2E0\uC624": 6,
  "\uC784\uC778": 6,
  "\uACC4\uC720": 6,
  "\uAC11\uC624": 7,
  "\uC744\uD574": 7,
  "\uBCD1\uC720": 7,
  "\uC815\uC778": 7,
  "\uBB34\uC720": 7,
  "\uAE30\uC778": 7,
  "\uACBD\uC790": 7,
  "\uC2E0\uC0AC": 7,
  "\uC784\uBB18": 7,
  "\uACC4\uC2E0": 7,
  "\uAC11\uBBF8": 8,
  "\uC744\uC220": 8,
  "\uBCD1\uC220": 8,
  "\uC815\uCD95": 8,
  "\uBB34\uC220": 8,
  "\uAE30\uCD95": 8,
  "\uACBD\uCD95": 8,
  "\uC2E0\uC9C4": 8,
  "\uC784\uC9C4": 8,
  "\uACC4\uBBF8": 8,
  "\uAC11\uC2E0": 9,
  "\uC744\uC720": 9,
  "\uBCD1\uD574": 9,
  "\uC815\uC790": 9,
  "\uBB34\uD574": 9,
  "\uAE30\uC790": 9,
  "\uACBD\uC778": 9,
  "\uC2E0\uBB18": 9,
  "\uC784\uC0AC": 9,
  "\uACC4\uC624": 9,
  "\uAC11\uC720": 10,
  "\uC744\uC2E0": 10,
  "\uBCD1\uC790": 10,
  "\uC815\uD574": 10,
  "\uBB34\uC790": 10,
  "\uAE30\uD574": 10,
  "\uACBD\uBB18": 10,
  "\uC2E0\uC778": 10,
  "\uC784\uC624": 10,
  "\uACC4\uC0AC": 10,
  "\uAC11\uC220": 11,
  "\uC744\uBBF8": 11,
  "\uBCD1\uCD95": 11,
  "\uC815\uC220": 11,
  "\uBB34\uCD95": 11,
  "\uAE30\uC220": 11,
  "\uACBD\uC9C4": 11,
  "\uC2E0\uCD95": 11,
  "\uC784\uBBF8": 11,
  "\uACC4\uC9C4": 11
};
var JIJANGGAN = {
  "\u5BC5": "\u620A\u4E19\u7532",
  "\u536F": "\u7532 \u4E59",
  "\u8FB0": "\u4E59\u7678\u620A",
  "\u5DF3": "\u620A\u5E9A\u4E19",
  "\u5348": "\u4E19\u5DF1\u4E01",
  "\u672A": "\u4E01\u4E59\u5DF1",
  "\u7533": "\u620A\u58EC\u5E9A",
  "\u9149": "\u5E9A \u8F9B",
  "\u620C": "\u8F9B\u4E01\u620A",
  "\u4EA5": "\u620A\u7532\u58EC",
  "\u5B50": "\u58EC \u7678",
  "\u4E11": "\u7678\u8F9B\u5DF1"
};
var GONGMANG_TABLE = [
  ["\u620C", "\u4EA5"],
  // 甲子旬 (0-9)
  ["\u7533", "\u9149"],
  // 甲戌旬 (10-19)
  ["\u5348", "\u672A"],
  // 甲申旬 (20-29)
  ["\u8FB0", "\u5DF3"],
  // 甲午旬 (30-39)
  ["\u5BC5", "\u536F"],
  // 甲辰旬 (40-49)
  ["\u5B50", "\u4E11"]
  // 甲寅旬 (50-59)
];
var HGANJI = [
  "\u7532\u5B50",
  "\u4E59\u4E11",
  "\u4E19\u5BC5",
  "\u4E01\u536F",
  "\u620A\u8FB0",
  "\u5DF1\u5DF3",
  "\u5E9A\u5348",
  "\u8F9B\u672A",
  "\u58EC\u7533",
  "\u7678\u9149",
  "\u7532\u620C",
  "\u4E59\u4EA5",
  "\u4E19\u5B50",
  "\u4E01\u4E11",
  "\u620A\u5BC5",
  "\u5DF1\u536F",
  "\u5E9A\u8FB0",
  "\u8F9B\u5DF3",
  "\u58EC\u5348",
  "\u7678\u672A",
  "\u7532\u7533",
  "\u4E59\u9149",
  "\u4E19\u620C",
  "\u4E01\u4EA5",
  "\u620A\u5B50",
  "\u5DF1\u4E11",
  "\u5E9A\u5BC5",
  "\u8F9B\u536F",
  "\u58EC\u8FB0",
  "\u7678\u5DF3",
  "\u7532\u5348",
  "\u4E59\u672A",
  "\u4E19\u7533",
  "\u4E01\u9149",
  "\u620A\u620C",
  "\u5DF1\u4EA5",
  "\u5E9A\u5B50",
  "\u8F9B\u4E11",
  "\u58EC\u5BC5",
  "\u7678\u536F",
  "\u7532\u8FB0",
  "\u4E59\u5DF3",
  "\u4E19\u5348",
  "\u4E01\u672A",
  "\u620A\u7533",
  "\u5DF1\u9149",
  "\u5E9A\u620C",
  "\u8F9B\u4EA5",
  "\u58EC\u5B50",
  "\u7678\u4E11",
  "\u7532\u5BC5",
  "\u4E59\u536F",
  "\u4E19\u8FB0",
  "\u4E01\u5DF3",
  "\u620A\u5348",
  "\u5DF1\u672A",
  "\u5E9A\u7533",
  "\u8F9B\u9149",
  "\u58EC\u620C",
  "\u7678\u4EA5"
];
var PILLAR_NAMES = ["\u6642\u67F1", "\u65E5\u67F1", "\u6708\u67F1", "\u5E74\u67F1"];
var TIAN_GAN = "\u7532\u4E59\u4E19\u4E01\u620A\u5DF1\u5E9A\u8F9B\u58EC\u7678";
var DI_ZHI = "\u5B50\u4E11\u5BC5\u536F\u8FB0\u5DF3\u5348\u672A\u7533\u9149\u620C\u4EA5";
var PALACE_NAMES = [
  "\u547D\u5BAE",
  "\u5144\u5F1F",
  "\u592B\u59BB",
  "\u5B50\u5973",
  "\u8CA1\u5E1B",
  "\u75BE\u5384",
  "\u9077\u79FB",
  "\u4EA4\u53CB",
  "\u5B98\u797F",
  "\u7530\u5B85",
  "\u798F\u5FB7",
  "\u7236\u6BCD"
];
var WU_XING_JU_MAP = {
  water: { name: "\u6C34\u4E8C\u5C40", number: 2 },
  tree: { name: "\u6728\u4E09\u5C40", number: 3 },
  metal: { name: "\u91D1\u56DB\u5C40", number: 4 },
  earth: { name: "\u571F\u4E94\u5C40", number: 5 },
  fire: { name: "\u706B\u516D\u5C40", number: 6 }
};
var MAIN_STAR_NAMES = /* @__PURE__ */ new Set([
  "\u7D2B\u5FAE",
  "\u5929\u6A5F",
  "\u592A\u967D",
  "\u6B66\u66F2",
  "\u5929\u540C",
  "\u5EC9\u8C9E",
  "\u5929\u5E9C",
  "\u592A\u9670",
  "\u8CAA\u72FC",
  "\u5DE8\u9580",
  "\u5929\u76F8",
  "\u5929\u6881",
  "\u4E03\u6BBA",
  "\u7834\u8ECD"
]);
var LUCKY_STAR_NAMES = /* @__PURE__ */ new Set([
  "\u5DE6\u8F14",
  "\u53F3\u5F3C",
  "\u6587\u660C",
  "\u6587\u66F2",
  "\u5929\u9B41",
  "\u5929\u925E",
  "\u797F\u5B58",
  "\u5929\u99AC"
]);
var SHA_STAR_NAMES = /* @__PURE__ */ new Set([
  "\u64CE\u7F8A",
  "\u9640\u7F85",
  "\u706B\u661F",
  "\u9234\u661F",
  "\u5730\u7A7A",
  "\u5730\u52AB"
]);
var NAYIN_TABLE = {
  "\u7532,\u5B50": "metal",
  "\u4E59,\u4E11": "metal",
  "\u4E19,\u5BC5": "fire",
  "\u4E01,\u536F": "fire",
  "\u620A,\u8FB0": "tree",
  "\u5DF1,\u5DF3": "tree",
  "\u5E9A,\u5348": "earth",
  "\u8F9B,\u672A": "earth",
  "\u58EC,\u7533": "metal",
  "\u7678,\u9149": "metal",
  "\u7532,\u620C": "fire",
  "\u4E59,\u4EA5": "fire",
  "\u4E19,\u5B50": "water",
  "\u4E01,\u4E11": "water",
  "\u620A,\u5BC5": "earth",
  "\u5DF1,\u536F": "earth",
  "\u5E9A,\u8FB0": "metal",
  "\u8F9B,\u5DF3": "metal",
  "\u58EC,\u5348": "tree",
  "\u7678,\u672A": "tree",
  "\u7532,\u7533": "water",
  "\u4E59,\u9149": "water",
  "\u4E19,\u620C": "earth",
  "\u4E01,\u4EA5": "earth",
  "\u620A,\u5B50": "fire",
  "\u5DF1,\u4E11": "fire",
  "\u5E9A,\u5BC5": "tree",
  "\u8F9B,\u536F": "tree",
  "\u58EC,\u8FB0": "water",
  "\u7678,\u5DF3": "water",
  "\u7532,\u5348": "metal",
  "\u4E59,\u672A": "metal",
  "\u4E19,\u7533": "fire",
  "\u4E01,\u9149": "fire",
  "\u620A,\u620C": "tree",
  "\u5DF1,\u4EA5": "tree",
  "\u5E9A,\u5B50": "earth",
  "\u8F9B,\u4E11": "earth",
  "\u58EC,\u5BC5": "metal",
  "\u7678,\u536F": "metal",
  "\u7532,\u8FB0": "fire",
  "\u4E59,\u5DF3": "fire",
  "\u4E19,\u5348": "water",
  "\u4E01,\u672A": "water",
  "\u620A,\u7533": "earth",
  "\u5DF1,\u9149": "earth",
  "\u5E9A,\u620C": "metal",
  "\u8F9B,\u4EA5": "metal",
  "\u58EC,\u5B50": "tree",
  "\u7678,\u4E11": "tree",
  "\u7532,\u5BC5": "water",
  "\u4E59,\u536F": "water",
  "\u4E19,\u8FB0": "earth",
  "\u4E01,\u5DF3": "earth",
  "\u620A,\u5348": "fire",
  "\u5DF1,\u672A": "fire",
  "\u5E9A,\u7533": "tree",
  "\u8F9B,\u9149": "tree",
  "\u58EC,\u620C": "water",
  "\u7678,\u4EA5": "water"
};
var SI_HUA_TABLE = {
  "\u7532": ["\u5EC9\u8C9E", "\u7834\u8ECD", "\u6B66\u66F2", "\u592A\u967D"],
  "\u4E59": ["\u5929\u6A5F", "\u5929\u6881", "\u7D2B\u5FAE", "\u592A\u9670"],
  "\u4E19": ["\u5929\u540C", "\u5929\u6A5F", "\u6587\u660C", "\u5EC9\u8C9E"],
  "\u4E01": ["\u592A\u9670", "\u5929\u540C", "\u5929\u6A5F", "\u5DE8\u9580"],
  "\u620A": ["\u8CAA\u72FC", "\u592A\u9670", "\u53F3\u5F3C", "\u5929\u6A5F"],
  "\u5DF1": ["\u6B66\u66F2", "\u8CAA\u72FC", "\u5929\u6881", "\u6587\u66F2"],
  "\u5E9A": ["\u592A\u967D", "\u6B66\u66F2", "\u592A\u9670", "\u5929\u540C"],
  "\u8F9B": ["\u5DE8\u9580", "\u592A\u967D", "\u6587\u66F2", "\u6587\u660C"],
  "\u58EC": ["\u5929\u6881", "\u7D2B\u5FAE", "\u5DE6\u8F14", "\u6B66\u66F2"],
  "\u7678": ["\u7834\u8ECD", "\u5DE8\u9580", "\u592A\u9670", "\u8CAA\u72FC"]
};
var LU_CUN_TABLE = {
  "\u7532": "\u5BC5",
  "\u4E59": "\u536F",
  "\u4E19": "\u5DF3",
  "\u4E01": "\u5348",
  "\u620A": "\u5DF3",
  "\u5DF1": "\u5348",
  "\u5E9A": "\u7533",
  "\u8F9B": "\u9149",
  "\u58EC": "\u4EA5",
  "\u7678": "\u5B50"
};
var KUI_YUE_TABLE = {
  "\u7532": ["\u4E11", "\u672A"],
  "\u620A": ["\u4E11", "\u672A"],
  "\u5E9A": ["\u4E11", "\u672A"],
  "\u4E59": ["\u5B50", "\u7533"],
  "\u5DF1": ["\u5B50", "\u7533"],
  "\u4E19": ["\u4EA5", "\u9149"],
  "\u4E01": ["\u4EA5", "\u9149"],
  "\u8F9B": ["\u5348", "\u5BC5"],
  "\u58EC": ["\u536F", "\u5DF3"],
  "\u7678": ["\u536F", "\u5DF3"]
};
var HUO_XING_START = {
  "\u5BC5": "\u4E11",
  "\u5348": "\u4E11",
  "\u620C": "\u4E11",
  "\u7533": "\u5BC5",
  "\u5B50": "\u5BC5",
  "\u8FB0": "\u5BC5",
  "\u5DF3": "\u536F",
  "\u9149": "\u536F",
  "\u4E11": "\u536F",
  "\u4EA5": "\u9149",
  "\u536F": "\u9149",
  "\u672A": "\u9149"
};
var LING_XING_START = {
  "\u5BC5": "\u536F",
  "\u5348": "\u536F",
  "\u620C": "\u536F",
  "\u7533": "\u620C",
  "\u5B50": "\u620C",
  "\u8FB0": "\u620C",
  "\u5DF3": "\u620C",
  "\u9149": "\u620C",
  "\u4E11": "\u620C",
  "\u4EA5": "\u620C",
  "\u536F": "\u620C",
  "\u672A": "\u620C"
};
var TIAN_MA_TABLE = {
  "\u5BC5": "\u7533",
  "\u5348": "\u7533",
  "\u620C": "\u7533",
  "\u7533": "\u5BC5",
  "\u5B50": "\u5BC5",
  "\u8FB0": "\u5BC5",
  "\u5DF3": "\u4EA5",
  "\u9149": "\u4EA5",
  "\u4E11": "\u4EA5",
  "\u4EA5": "\u5DF3",
  "\u536F": "\u5DF3",
  "\u672A": "\u5DF3"
};
var ZIWEI_SERIES_OFFSETS = {
  "\u7D2B\u5FAE": 0,
  "\u5929\u6A5F": -1,
  "\u592A\u967D": -3,
  "\u6B66\u66F2": -4,
  "\u5929\u540C": -5,
  "\u5EC9\u8C9E": -8
};
var TIANFU_SERIES_OFFSETS = {
  "\u5929\u5E9C": 0,
  "\u592A\u9670": 1,
  "\u8CAA\u72FC": 2,
  "\u5DE8\u9580": 3,
  "\u5929\u76F8": 4,
  "\u5929\u6881": 5,
  "\u4E03\u6BBA": 6,
  "\u7834\u8ECD": 10
};
var BRIGHTNESS_TABLE = {
  "\u7D2B\u5FAE": {
    "\u5B50": "\u65FA",
    "\u4E11": "\u5EDF",
    "\u5BC5": "\u5F97",
    "\u536F": "\u5229",
    "\u8FB0": "\u65FA",
    "\u5DF3": "\u5F97",
    "\u5348": "\u65FA",
    "\u672A": "\u5EDF",
    "\u7533": "\u5F97",
    "\u9149": "\u5229",
    "\u620C": "\u65FA",
    "\u4EA5": "\u5F97"
  },
  "\u5929\u6A5F": {
    "\u5B50": "\u5EDF",
    "\u4E11": "\u9677",
    "\u5BC5": "\u65FA",
    "\u536F": "\u5EDF",
    "\u8FB0": "\u5F97",
    "\u5DF3": "\u9677",
    "\u5348": "\u9677",
    "\u672A": "\u9677",
    "\u7533": "\u65FA",
    "\u9149": "\u5EDF",
    "\u620C": "\u5F97",
    "\u4EA5": "\u9677"
  },
  "\u592A\u967D": {
    "\u5B50": "\u9677",
    "\u4E11": "\u9677",
    "\u5BC5": "\u65FA",
    "\u536F": "\u5EDF",
    "\u8FB0": "\u65FA",
    "\u5DF3": "\u5EDF",
    "\u5348": "\u65FA",
    "\u672A": "\u5F97",
    "\u7533": "\u5F97",
    "\u9149": "\u9677",
    "\u620C": "\u9677",
    "\u4EA5": "\u9677"
  },
  "\u6B66\u66F2": {
    "\u5B50": "\u65FA",
    "\u4E11": "\u5EDF",
    "\u5BC5": "\u5229",
    "\u536F": "\u9677",
    "\u8FB0": "\u5EDF",
    "\u5DF3": "\u5229",
    "\u5348": "\u65FA",
    "\u672A": "\u5EDF",
    "\u7533": "\u5F97",
    "\u9149": "\u9677",
    "\u620C": "\u5EDF",
    "\u4EA5": "\u5F97"
  },
  "\u5929\u540C": {
    "\u5B50": "\u65FA",
    "\u4E11": "\u9677",
    "\u5BC5": "\u5F97",
    "\u536F": "\u9677",
    "\u8FB0": "\u9677",
    "\u5DF3": "\u65FA",
    "\u5348": "\u9677",
    "\u672A": "\u9677",
    "\u7533": "\u5F97",
    "\u9149": "\u9677",
    "\u620C": "\u9677",
    "\u4EA5": "\u65FA"
  },
  "\u5EC9\u8C9E": {
    "\u5B50": "\u5E73",
    "\u4E11": "\u5EDF",
    "\u5BC5": "\u5EDF",
    "\u536F": "\u5E73",
    "\u8FB0": "\u5E73",
    "\u5DF3": "\u9677",
    "\u5348": "\u5E73",
    "\u672A": "\u5EDF",
    "\u7533": "\u5EDF",
    "\u9149": "\u5E73",
    "\u620C": "\u5E73",
    "\u4EA5": "\u9677"
  },
  "\u5929\u5E9C": {
    "\u5B50": "\u5EDF",
    "\u4E11": "\u65FA",
    "\u5BC5": "\u5F97",
    "\u536F": "\u5EDF",
    "\u8FB0": "\u65FA",
    "\u5DF3": "\u5EDF",
    "\u5348": "\u5EDF",
    "\u672A": "\u65FA",
    "\u7533": "\u5F97",
    "\u9149": "\u5EDF",
    "\u620C": "\u65FA",
    "\u4EA5": "\u5EDF"
  },
  "\u592A\u9670": {
    "\u5B50": "\u5EDF",
    "\u4E11": "\u65FA",
    "\u5BC5": "\u9677",
    "\u536F": "\u9677",
    "\u8FB0": "\u9677",
    "\u5DF3": "\u9677",
    "\u5348": "\u9677",
    "\u672A": "\u9677",
    "\u7533": "\u5F97",
    "\u9149": "\u65FA",
    "\u620C": "\u5EDF",
    "\u4EA5": "\u5EDF"
  },
  "\u8CAA\u72FC": {
    "\u5B50": "\u65FA",
    "\u4E11": "\u5EDF",
    "\u5BC5": "\u5EDF",
    "\u536F": "\u9677",
    "\u8FB0": "\u65FA",
    "\u5DF3": "\u9677",
    "\u5348": "\u65FA",
    "\u672A": "\u5EDF",
    "\u7533": "\u5EDF",
    "\u9149": "\u9677",
    "\u620C": "\u65FA",
    "\u4EA5": "\u9677"
  },
  "\u5DE8\u9580": {
    "\u5B50": "\u65FA",
    "\u4E11": "\u9677",
    "\u5BC5": "\u5EDF",
    "\u536F": "\u65FA",
    "\u8FB0": "\u9677",
    "\u5DF3": "\u9677",
    "\u5348": "\u65FA",
    "\u672A": "\u9677",
    "\u7533": "\u5EDF",
    "\u9149": "\u65FA",
    "\u620C": "\u9677",
    "\u4EA5": "\u9677"
  },
  "\u5929\u76F8": {
    "\u5B50": "\u5EDF",
    "\u4E11": "\u5EDF",
    "\u5BC5": "\u65FA",
    "\u536F": "\u5229",
    "\u8FB0": "\u5F97",
    "\u5DF3": "\u5EDF",
    "\u5348": "\u5EDF",
    "\u672A": "\u5EDF",
    "\u7533": "\u65FA",
    "\u9149": "\u5229",
    "\u620C": "\u5F97",
    "\u4EA5": "\u5EDF"
  },
  "\u5929\u6881": {
    "\u5B50": "\u5EDF",
    "\u4E11": "\u65FA",
    "\u5BC5": "\u5EDF",
    "\u536F": "\u65FA",
    "\u8FB0": "\u9677",
    "\u5DF3": "\u5EDF",
    "\u5348": "\u5EDF",
    "\u672A": "\u65FA",
    "\u7533": "\u5EDF",
    "\u9149": "\u65FA",
    "\u620C": "\u9677",
    "\u4EA5": "\u5EDF"
  },
  "\u4E03\u6BBA": {
    "\u5B50": "\u65FA",
    "\u4E11": "\u5EDF",
    "\u5BC5": "\u5EDF",
    "\u536F": "\u65FA",
    "\u8FB0": "\u5F97",
    "\u5DF3": "\u5EDF",
    "\u5348": "\u65FA",
    "\u672A": "\u5EDF",
    "\u7533": "\u5EDF",
    "\u9149": "\u65FA",
    "\u620C": "\u5F97",
    "\u4EA5": "\u5EDF"
  },
  "\u7834\u8ECD": {
    "\u5B50": "\u65FA",
    "\u4E11": "\u5EDF",
    "\u5BC5": "\u9677",
    "\u536F": "\u65FA",
    "\u8FB0": "\u9677",
    "\u5DF3": "\u5EDF",
    "\u5348": "\u65FA",
    "\u672A": "\u5EDF",
    "\u7533": "\u9677",
    "\u9149": "\u65FA",
    "\u620C": "\u9677",
    "\u4EA5": "\u5EDF"
  }
};
var CHANGSHENG_START = {
  water2: "\u7533",
  wood3: "\u4EA5",
  metal4: "\u5DF3",
  earth5: "\u7533",
  fire6: "\u5BC5"
};
var WU_HU_DUN_GAN = {
  "\u7532": "\u4E19",
  "\u5DF1": "\u4E19",
  "\u4E59": "\u620A",
  "\u5E9A": "\u620A",
  "\u4E19": "\u5E9A",
  "\u8F9B": "\u5E9A",
  "\u4E01": "\u58EC",
  "\u58EC": "\u58EC",
  "\u620A": "\u7532",
  "\u7678": "\u7532"
};

export {
  SKY,
  EARTH,
  SKY_KR,
  EARTH_KR,
  YANGGAN,
  RELATIONS,
  METEORS_12,
  SPIRITS_12,
  BRANCH_ELEMENT,
  STEM_INFO,
  ELEMENT_HANJA,
  TRIPLE_COMPOSES,
  TRIPLE_COMPOSE_ELEMENTS,
  HALF_COMPOSES,
  DIRECTIONAL_COMPOSES,
  DIRECTIONAL_COMPOSE_ELEMENTS,
  STEM_COMBINES,
  STEM_CLASHES,
  BRANCH_COMBINES_6,
  BRANCH_CLASHES,
  BRANCH_BREAKS,
  BRANCH_HARMS,
  BRANCH_PUNISHMENTS,
  BRANCH_SELF_PUNISHMENTS,
  BRANCH_WONJIN,
  BRANCH_GWIMUN,
  YANGIN_MAP,
  BAEKHO_PILLARS,
  GOEGANG_PILLARS,
  DOHWA_MAP,
  CHEONUL_MAP,
  CHEONDUK_MAP,
  WOLDUK_MAP,
  MUNCHANG_MAP,
  HONGYEOM_PILLARS,
  GEUMYEO_MAP,
  METEOR_LOOKUP,
  JIJANGGAN,
  GONGMANG_TABLE,
  HGANJI,
  PILLAR_NAMES,
  TIAN_GAN,
  DI_ZHI,
  PALACE_NAMES,
  WU_XING_JU_MAP,
  MAIN_STAR_NAMES,
  LUCKY_STAR_NAMES,
  SHA_STAR_NAMES,
  NAYIN_TABLE,
  SI_HUA_TABLE,
  LU_CUN_TABLE,
  KUI_YUE_TABLE,
  HUO_XING_START,
  LING_XING_START,
  TIAN_MA_TABLE,
  ZIWEI_SERIES_OFFSETS,
  TIANFU_SERIES_OFFSETS,
  BRIGHTNESS_TABLE,
  CHANGSHENG_START,
  WU_HU_DUN_GAN
};
