// @orrery/core/dist/chunk-WOQT7EBV.js
// src/cities.ts
var JAMO_TO_INITIAL = {
  12593: 0,
  12594: 1,
  12596: 2,
  12599: 3,
  12600: 4,
  12601: 5,
  12609: 6,
  12610: 7,
  12611: 8,
  12613: 9,
  12614: 10,
  12615: 11,
  12616: 12,
  12617: 13,
  12618: 14,
  12619: 15,
  12620: 16,
  12621: 17,
  12622: 18
};
function getInitialIndex(char) {
  const code = char.charCodeAt(0);
  if (code >= 44032 && code <= 55203) {
    return Math.floor((code - 44032) / (21 * 28));
  }
  if (code in JAMO_TO_INITIAL) {
    return JAMO_TO_INITIAL[code];
  }
  return null;
}
function isJamo(char) {
  return char.charCodeAt(0) in JAMO_TO_INITIAL;
}
function isAllChosung(query) {
  return query.length > 0 && [...query].every(isJamo);
}
function matchChosung(name, query) {
  if (query.length > name.length) return false;
  for (let i = 0; i < query.length; i++) {
    const qIdx = JAMO_TO_INITIAL[query.charCodeAt(i)];
    const nIdx = getInitialIndex(name[i]);
    if (qIdx === void 0 || nIdx === null || qIdx !== nIdx) return false;
  }
  return true;
}
var KOREAN_CITIES = [
  // 특별/광역시
  { name: "\uC11C\uC6B8", region: "\uC11C\uC6B8\uD2B9\uBCC4\uC2DC", lat: 37.5665, lon: 126.978 },
  { name: "\uBD80\uC0B0", region: "\uBD80\uC0B0\uAD11\uC5ED\uC2DC", lat: 35.1796, lon: 129.0756 },
  { name: "\uC778\uCC9C", region: "\uC778\uCC9C\uAD11\uC5ED\uC2DC", lat: 37.4563, lon: 126.7052 },
  { name: "\uB300\uAD6C", region: "\uB300\uAD6C\uAD11\uC5ED\uC2DC", lat: 35.8714, lon: 128.6014 },
  { name: "\uB300\uC804", region: "\uB300\uC804\uAD11\uC5ED\uC2DC", lat: 36.3504, lon: 127.3845 },
  { name: "\uAD11\uC8FC", region: "\uAD11\uC8FC\uAD11\uC5ED\uC2DC", lat: 35.1595, lon: 126.8526 },
  { name: "\uC6B8\uC0B0", region: "\uC6B8\uC0B0\uAD11\uC5ED\uC2DC", lat: 35.5384, lon: 129.3114 },
  { name: "\uC138\uC885", region: "\uC138\uC885\uD2B9\uBCC4\uC790\uCE58\uC2DC", lat: 36.48, lon: 127.259 },
  // 경기도
  { name: "\uC218\uC6D0", region: "\uACBD\uAE30\uB3C4", lat: 37.2636, lon: 127.0286 },
  { name: "\uACE0\uC591", region: "\uACBD\uAE30\uB3C4", lat: 37.6584, lon: 126.832 },
  { name: "\uC6A9\uC778", region: "\uACBD\uAE30\uB3C4", lat: 37.2411, lon: 127.1776 },
  { name: "\uC131\uB0A8", region: "\uACBD\uAE30\uB3C4", lat: 37.4386, lon: 127.1378 },
  { name: "\uBD80\uCC9C", region: "\uACBD\uAE30\uB3C4", lat: 37.5034, lon: 126.766 },
  { name: "\uC548\uC0B0", region: "\uACBD\uAE30\uB3C4", lat: 37.3219, lon: 126.8309 },
  { name: "\uC548\uC591", region: "\uACBD\uAE30\uB3C4", lat: 37.3943, lon: 126.9568 },
  { name: "\uB0A8\uC591\uC8FC", region: "\uACBD\uAE30\uB3C4", lat: 37.636, lon: 127.2165 },
  { name: "\uD654\uC131", region: "\uACBD\uAE30\uB3C4", lat: 37.1996, lon: 126.8312 },
  { name: "\uD3C9\uD0DD", region: "\uACBD\uAE30\uB3C4", lat: 36.9921, lon: 127.0857 },
  { name: "\uC758\uC815\uBD80", region: "\uACBD\uAE30\uB3C4", lat: 37.7381, lon: 127.0338 },
  { name: "\uC2DC\uD765", region: "\uACBD\uAE30\uB3C4", lat: 37.3801, lon: 126.8032 },
  { name: "\uD30C\uC8FC", region: "\uACBD\uAE30\uB3C4", lat: 37.759, lon: 126.7802 },
  { name: "\uAE40\uD3EC", region: "\uACBD\uAE30\uB3C4", lat: 37.6153, lon: 126.7156 },
  { name: "\uAD11\uBA85", region: "\uACBD\uAE30\uB3C4", lat: 37.4786, lon: 126.8646 },
  { name: "\uC774\uCC9C", region: "\uACBD\uAE30\uB3C4", lat: 37.272, lon: 127.435 },
  { name: "\uAD70\uD3EC", region: "\uACBD\uAE30\uB3C4", lat: 37.3616, lon: 126.9351 },
  { name: "\uD558\uB0A8", region: "\uACBD\uAE30\uB3C4", lat: 37.5393, lon: 127.2148 },
  { name: "\uC624\uC0B0", region: "\uACBD\uAE30\uB3C4", lat: 37.1499, lon: 127.077 },
  { name: "\uC591\uC8FC", region: "\uACBD\uAE30\uB3C4", lat: 37.7853, lon: 127.0456 },
  { name: "\uAD6C\uB9AC", region: "\uACBD\uAE30\uB3C4", lat: 37.5943, lon: 127.1297 },
  { name: "\uC548\uC131", region: "\uACBD\uAE30\uB3C4", lat: 37.008, lon: 127.2798 },
  { name: "\uD3EC\uCC9C", region: "\uACBD\uAE30\uB3C4", lat: 37.8949, lon: 127.2003 },
  { name: "\uC758\uC655", region: "\uACBD\uAE30\uB3C4", lat: 37.3449, lon: 126.9685 },
  { name: "\uC5EC\uC8FC", region: "\uACBD\uAE30\uB3C4", lat: 37.2984, lon: 127.6373 },
  { name: "\uC591\uD3C9", region: "\uACBD\uAE30\uB3C4", lat: 37.4914, lon: 127.4875 },
  { name: "\uB3D9\uB450\uCC9C", region: "\uACBD\uAE30\uB3C4", lat: 37.9034, lon: 127.0606 },
  { name: "\uACFC\uCC9C", region: "\uACBD\uAE30\uB3C4", lat: 37.4292, lon: 126.9876 },
  { name: "\uAD11\uC8FC", region: "\uACBD\uAE30\uB3C4", lat: 37.4095, lon: 127.255 },
  { name: "\uC5F0\uCC9C", region: "\uACBD\uAE30\uB3C4", lat: 38.0965, lon: 127.075 },
  { name: "\uAC00\uD3C9", region: "\uACBD\uAE30\uB3C4", lat: 37.8315, lon: 127.5095 },
  // 강원도
  { name: "\uCD98\uCC9C", region: "\uAC15\uC6D0\uB3C4", lat: 37.8813, lon: 127.7299 },
  { name: "\uC6D0\uC8FC", region: "\uAC15\uC6D0\uB3C4", lat: 37.3422, lon: 127.9202 },
  { name: "\uAC15\uB989", region: "\uAC15\uC6D0\uB3C4", lat: 37.7519, lon: 128.8761 },
  { name: "\uC18D\uCD08", region: "\uAC15\uC6D0\uB3C4", lat: 38.207, lon: 128.5918 },
  // 충청도
  { name: "\uCCAD\uC8FC", region: "\uCDA9\uCCAD\uBD81\uB3C4", lat: 36.6424, lon: 127.489 },
  { name: "\uCC9C\uC548", region: "\uCDA9\uCCAD\uB0A8\uB3C4", lat: 36.8152, lon: 127.1139 },
  { name: "\uCDA9\uC8FC", region: "\uCDA9\uCCAD\uBD81\uB3C4", lat: 36.991, lon: 127.9259 },
  { name: "\uC81C\uCC9C", region: "\uCDA9\uCCAD\uBD81\uB3C4", lat: 37.1326, lon: 128.191 },
  { name: "\uC544\uC0B0", region: "\uCDA9\uCCAD\uB0A8\uB3C4", lat: 36.7898, lon: 127.0018 },
  // 전라도
  { name: "\uC804\uC8FC", region: "\uC804\uB77C\uBD81\uB3C4", lat: 35.8242, lon: 127.148 },
  { name: "\uBAA9\uD3EC", region: "\uC804\uB77C\uB0A8\uB3C4", lat: 34.8118, lon: 126.3922 },
  { name: "\uC5EC\uC218", region: "\uC804\uB77C\uB0A8\uB3C4", lat: 34.7604, lon: 127.6622 },
  { name: "\uC21C\uCC9C", region: "\uC804\uB77C\uB0A8\uB3C4", lat: 34.9505, lon: 127.4873 },
  { name: "\uAD70\uC0B0", region: "\uC804\uB77C\uBD81\uB3C4", lat: 35.9677, lon: 126.737 },
  { name: "\uC775\uC0B0", region: "\uC804\uB77C\uBD81\uB3C4", lat: 35.9483, lon: 126.9576 },
  // 경상도
  { name: "\uD3EC\uD56D", region: "\uACBD\uC0C1\uBD81\uB3C4", lat: 36.019, lon: 129.3435 },
  { name: "\uACBD\uC8FC", region: "\uACBD\uC0C1\uBD81\uB3C4", lat: 35.8562, lon: 129.2247 },
  { name: "\uAE40\uD574", region: "\uACBD\uC0C1\uB0A8\uB3C4", lat: 35.2285, lon: 128.8894 },
  { name: "\uCC3D\uC6D0", region: "\uACBD\uC0C1\uB0A8\uB3C4", lat: 35.2281, lon: 128.6811 },
  { name: "\uC9C4\uC8FC", region: "\uACBD\uC0C1\uB0A8\uB3C4", lat: 35.1798, lon: 128.1076 },
  { name: "\uAD6C\uBBF8", region: "\uACBD\uC0C1\uBD81\uB3C4", lat: 36.1197, lon: 128.3446 },
  { name: "\uC548\uB3D9", region: "\uACBD\uC0C1\uBD81\uB3C4", lat: 36.5684, lon: 128.7294 },
  // 제주도
  { name: "\uC81C\uC8FC", region: "\uC81C\uC8FC\uD2B9\uBCC4\uC790\uCE58\uB3C4", lat: 33.4996, lon: 126.5312 },
  { name: "\uC11C\uADC0\uD3EC", region: "\uC81C\uC8FC\uD2B9\uBCC4\uC790\uCE58\uB3C4", lat: 33.2542, lon: 126.56 }
];
var WORLD_CITIES = [
  // ── 동아시아 ──
  { name: "\uB3C4\uCFC4", country: "\uC77C\uBCF8", lat: 35.6762, lon: 139.6503 },
  { name: "\uC624\uC0AC\uCE74", country: "\uC77C\uBCF8", lat: 34.6937, lon: 135.5023 },
  { name: "\uAD50\uD1A0", country: "\uC77C\uBCF8", lat: 35.0116, lon: 135.7681 },
  { name: "\uD6C4\uCFE0\uC624\uCE74", country: "\uC77C\uBCF8", lat: 33.5904, lon: 130.4017 },
  { name: "\uC0BF\uD3EC\uB85C", country: "\uC77C\uBCF8", lat: 43.0618, lon: 141.3545 },
  { name: "\uB098\uACE0\uC57C", country: "\uC77C\uBCF8", lat: 35.1815, lon: 136.9066 },
  { name: "\uBCA0\uC774\uC9D5", country: "\uC911\uAD6D", lat: 39.9042, lon: 116.4074 },
  { name: "\uC0C1\uD558\uC774", country: "\uC911\uAD6D", lat: 31.2304, lon: 121.4737 },
  { name: "\uAD11\uC800\uC6B0", country: "\uC911\uAD6D", lat: 23.1291, lon: 113.2644 },
  { name: "\uC120\uC804", country: "\uC911\uAD6D", lat: 22.5431, lon: 114.0579 },
  { name: "\uCCAD\uB450", country: "\uC911\uAD6D", lat: 30.5728, lon: 104.0668 },
  { name: "\uCDA9\uCE6D", country: "\uC911\uAD6D", lat: 29.4316, lon: 106.9123 },
  { name: "\uC2DC\uC548", country: "\uC911\uAD6D", lat: 34.3416, lon: 108.9398 },
  { name: "\uD64D\uCF69", country: "\uC911\uAD6D", lat: 22.3193, lon: 114.1694 },
  { name: "\uD0C0\uC774\uBCA0\uC774", country: "\uB300\uB9CC", lat: 25.033, lon: 121.5654 },
  { name: "\uC6B8\uB780\uBC14\uD1A0\uB974", country: "\uBABD\uACE8", lat: 47.8864, lon: 106.9057 },
  { name: "\uD3C9\uC591", country: "\uBD81\uD55C", lat: 39.0392, lon: 125.7625 },
  // ── 동남아시아 ──
  { name: "\uBC29\uCF55", country: "\uD0DC\uAD6D", lat: 13.7563, lon: 100.5018 },
  { name: "\uD558\uB178\uC774", country: "\uBCA0\uD2B8\uB0A8", lat: 21.0278, lon: 105.8342 },
  { name: "\uD638\uCE58\uBBFC", country: "\uBCA0\uD2B8\uB0A8", lat: 10.8231, lon: 106.6297 },
  { name: "\uB2E4\uB0AD", country: "\uBCA0\uD2B8\uB0A8", lat: 16.0544, lon: 108.2022 },
  { name: "\uC790\uCE74\uB974\uD0C0", country: "\uC778\uB3C4\uB124\uC2DC\uC544", lat: -6.2088, lon: 106.8456 },
  { name: "\uBC1C\uB9AC", country: "\uC778\uB3C4\uB124\uC2DC\uC544", lat: -8.3405, lon: 115.092 },
  { name: "\uC2F1\uAC00\uD3EC\uB974", country: "\uC2F1\uAC00\uD3EC\uB974", lat: 1.3521, lon: 103.8198 },
  { name: "\uCFE0\uC54C\uB77C\uB8F8\uD478\uB974", country: "\uB9D0\uB808\uC774\uC2DC\uC544", lat: 3.139, lon: 101.6869 },
  { name: "\uB9C8\uB2D0\uB77C", country: "\uD544\uB9AC\uD540", lat: 14.5995, lon: 120.9842 },
  { name: "\uC138\uBD80", country: "\uD544\uB9AC\uD540", lat: 10.3157, lon: 123.8854 },
  { name: "\uC591\uACE4", country: "\uBBF8\uC580\uB9C8", lat: 16.8661, lon: 96.1951 },
  { name: "\uD504\uB188\uD39C", country: "\uCE84\uBCF4\uB514\uC544", lat: 11.5564, lon: 104.9282 },
  { name: "\uBE44\uC5D4\uD2F0\uC548", country: "\uB77C\uC624\uC2A4", lat: 17.9757, lon: 102.6331 },
  // ── 남아시아 ──
  { name: "\uB274\uB378\uB9AC", country: "\uC778\uB3C4", lat: 28.6139, lon: 77.209 },
  { name: "\uBB44\uBC14\uC774", country: "\uC778\uB3C4", lat: 19.076, lon: 72.8777 },
  { name: "\uBCB5\uAC08\uB8E8\uB8E8", country: "\uC778\uB3C4", lat: 12.9716, lon: 77.5946 },
  { name: "\uB2E4\uCE74", country: "\uBC29\uAE00\uB77C\uB370\uC2DC", lat: 23.8103, lon: 90.4125 },
  { name: "\uC774\uC2AC\uB77C\uB9C8\uBC14\uB4DC", country: "\uD30C\uD0A4\uC2A4\uD0C4", lat: 33.6844, lon: 73.0479 },
  { name: "\uCE74\uB77C\uCE58", country: "\uD30C\uD0A4\uC2A4\uD0C4", lat: 24.8607, lon: 67.0011 },
  { name: "\uCE74\uD2B8\uB9CC\uB450", country: "\uB124\uD314", lat: 27.7172, lon: 85.324 },
  { name: "\uCF5C\uB86C\uBCF4", country: "\uC2A4\uB9AC\uB791\uCE74", lat: 6.9271, lon: 79.8612 },
  // ── 중앙아시아 / 서아시아 ──
  { name: "\uC544\uC2A4\uD0C0\uB098", country: "\uCE74\uC790\uD750\uC2A4\uD0C4", lat: 51.1694, lon: 71.4491 },
  { name: "\uC54C\uB9C8\uD2F0", country: "\uCE74\uC790\uD750\uC2A4\uD0C4", lat: 43.222, lon: 76.8512 },
  { name: "\uD0C0\uC288\uCF04\uD2B8", country: "\uC6B0\uC988\uBCA0\uD0A4\uC2A4\uD0C4", lat: 41.2995, lon: 69.2401 },
  { name: "\uBE44\uC288\uCF00\uD06C", country: "\uD0A4\uB974\uAE30\uC2A4\uC2A4\uD0C4", lat: 42.8746, lon: 74.5698 },
  { name: "\uB450\uC0E8\uBCA0", country: "\uD0C0\uC9C0\uD0A4\uC2A4\uD0C4", lat: 38.5598, lon: 68.7738 },
  { name: "\uC544\uC2DC\uAC00\uBC14\uD2B8", country: "\uD22C\uB974\uD06C\uBA54\uB2C8\uC2A4\uD0C4", lat: 37.9601, lon: 58.3261 },
  { name: "\uD14C\uD5E4\uB780", country: "\uC774\uB780", lat: 35.6892, lon: 51.389 },
  { name: "\uBC14\uADF8\uB2E4\uB4DC", country: "\uC774\uB77C\uD06C", lat: 33.3128, lon: 44.3615 },
  { name: "\uB9AC\uC57C\uB4DC", country: "\uC0AC\uC6B0\uB514\uC544\uB77C\uBE44\uC544", lat: 24.7136, lon: 46.6753 },
  { name: "\uC81C\uB2E4", country: "\uC0AC\uC6B0\uB514\uC544\uB77C\uBE44\uC544", lat: 21.4858, lon: 39.1925 },
  { name: "\uB450\uBC14\uC774", country: "\uC544\uB78D\uC5D0\uBBF8\uB9AC\uD2B8", lat: 25.2048, lon: 55.2708 },
  { name: "\uC544\uBD80\uB2E4\uBE44", country: "\uC544\uB78D\uC5D0\uBBF8\uB9AC\uD2B8", lat: 24.4539, lon: 54.3773 },
  { name: "\uB3C4\uD558", country: "\uCE74\uD0C0\uB974", lat: 25.2854, lon: 51.531 },
  { name: "\uCFE0\uC6E8\uC774\uD2B8\uC2DC\uD2F0", country: "\uCFE0\uC6E8\uC774\uD2B8", lat: 29.3759, lon: 47.9774 },
  { name: "\uC559\uCE74\uB77C", country: "\uD280\uB974\uD0A4\uC608", lat: 39.9334, lon: 32.8597 },
  { name: "\uC774\uC2A4\uD0C4\uBD88", country: "\uD280\uB974\uD0A4\uC608", lat: 41.0082, lon: 28.9784 },
  { name: "\uD154\uC544\uBE44\uBE0C", country: "\uC774\uC2A4\uB77C\uC5D8", lat: 32.0853, lon: 34.7818 },
  { name: "\uC608\uB8E8\uC0B4\uB818", country: "\uC774\uC2A4\uB77C\uC5D8", lat: 31.7683, lon: 35.2137 },
  { name: "\uBCA0\uC774\uB8E8\uD2B8", country: "\uB808\uBC14\uB17C", lat: 33.8938, lon: 35.5018 },
  { name: "\uC554\uB9CC", country: "\uC694\uB974\uB2E8", lat: 31.9454, lon: 35.9284 },
  { name: "\uB2E4\uB9C8\uC2A4\uCFE0\uC2A4", country: "\uC2DC\uB9AC\uC544", lat: 33.5138, lon: 36.2765 },
  { name: "\uD2B8\uBE4C\uB9AC\uC2DC", country: "\uC870\uC9C0\uC544", lat: 41.7151, lon: 44.8271 },
  { name: "\uBC14\uCFE0", country: "\uC544\uC81C\uB974\uBC14\uC774\uC794", lat: 40.4093, lon: 49.8671 },
  { name: "\uC608\uB808\uBC18", country: "\uC544\uB974\uBA54\uB2C8\uC544", lat: 40.1792, lon: 44.4991 },
  // ── 서유럽 ──
  { name: "\uB7F0\uB358", country: "\uC601\uAD6D", lat: 51.5074, lon: -0.1278 },
  { name: "\uC5D0\uB4E0\uBC84\uB7EC", country: "\uC601\uAD6D", lat: 55.9533, lon: -3.1883 },
  { name: "\uB9E8\uCCB4\uC2A4\uD130", country: "\uC601\uAD6D", lat: 53.4808, lon: -2.2426 },
  { name: "\uD30C\uB9AC", country: "\uD504\uB791\uC2A4", lat: 48.8566, lon: 2.3522 },
  { name: "\uB9C8\uB974\uC138\uC720", country: "\uD504\uB791\uC2A4", lat: 43.2965, lon: 5.3698 },
  { name: "\uB9AC\uC639", country: "\uD504\uB791\uC2A4", lat: 45.764, lon: 4.8357 },
  { name: "\uBCA0\uB97C\uB9B0", country: "\uB3C5\uC77C", lat: 52.52, lon: 13.405 },
  { name: "\uBB8C\uD5E8", country: "\uB3C5\uC77C", lat: 48.1351, lon: 11.582 },
  { name: "\uD568\uBD80\uB974\uD06C", country: "\uB3C5\uC77C", lat: 53.5511, lon: 9.9937 },
  { name: "\uD504\uB791\uD06C\uD478\uB974\uD2B8", country: "\uB3C5\uC77C", lat: 50.1109, lon: 8.6821 },
  { name: "\uC554\uC2A4\uD14C\uB974\uB2F4", country: "\uB124\uB35C\uB780\uB4DC", lat: 52.3676, lon: 4.9041 },
  { name: "\uBE0C\uB93C\uC140", country: "\uBCA8\uAE30\uC5D0", lat: 50.8503, lon: 4.3517 },
  { name: "\uB8E9\uC148\uBD80\uB974\uD06C", country: "\uB8E9\uC148\uBD80\uB974\uD06C", lat: 49.6117, lon: 6.1319 },
  // ── 남유럽 ──
  { name: "\uB9C8\uB4DC\uB9AC\uB4DC", country: "\uC2A4\uD398\uC778", lat: 40.4168, lon: -3.7038 },
  { name: "\uBC14\uB974\uC140\uB85C\uB098", country: "\uC2A4\uD398\uC778", lat: 41.3874, lon: 2.1686 },
  { name: "\uB85C\uB9C8", country: "\uC774\uD0C8\uB9AC\uC544", lat: 41.9028, lon: 12.4964 },
  { name: "\uBC00\uB77C\uB178", country: "\uC774\uD0C8\uB9AC\uC544", lat: 45.4642, lon: 9.19 },
  { name: "\uB098\uD3F4\uB9AC", country: "\uC774\uD0C8\uB9AC\uC544", lat: 40.8518, lon: 14.2681 },
  { name: "\uB9AC\uC2A4\uBCF8", country: "\uD3EC\uB974\uD22C\uAC08", lat: 38.7223, lon: -9.1393 },
  { name: "\uC544\uD14C\uB124", country: "\uADF8\uB9AC\uC2A4", lat: 37.9838, lon: 23.7275 },
  // ── 중유럽 ──
  { name: "\uBE48", country: "\uC624\uC2A4\uD2B8\uB9AC\uC544", lat: 48.2082, lon: 16.3738 },
  { name: "\uBCA0\uB978", country: "\uC2A4\uC704\uC2A4", lat: 46.948, lon: 7.4474 },
  { name: "\uCDE8\uB9AC\uD788", country: "\uC2A4\uC704\uC2A4", lat: 47.3769, lon: 8.5417 },
  { name: "\uC81C\uB124\uBC14", country: "\uC2A4\uC704\uC2A4", lat: 46.2044, lon: 6.1432 },
  { name: "\uD504\uB77C\uD558", country: "\uCCB4\uCF54", lat: 50.0755, lon: 14.4378 },
  { name: "\uBC14\uB974\uC0E4\uBC14", country: "\uD3F4\uB780\uB4DC", lat: 52.2297, lon: 21.0122 },
  { name: "\uD06C\uB77C\uCFE0\uD504", country: "\uD3F4\uB780\uB4DC", lat: 50.0647, lon: 19.945 },
  { name: "\uBD80\uB2E4\uD398\uC2A4\uD2B8", country: "\uD5DD\uAC00\uB9AC", lat: 47.4979, lon: 19.0402 },
  { name: "\uBD80\uCFE0\uB808\uC288\uD2F0", country: "\uB8E8\uB9C8\uB2C8\uC544", lat: 44.4268, lon: 26.1025 },
  { name: "\uC18C\uD53C\uC544", country: "\uBD88\uAC00\uB9AC\uC544", lat: 42.6977, lon: 23.3219 },
  // ── 발칸/남동유럽 ──
  { name: "\uC790\uADF8\uB808\uBE0C", country: "\uD06C\uB85C\uC544\uD2F0\uC544", lat: 45.815, lon: 15.9819 },
  { name: "\uB958\uBE14\uB7B4\uB098", country: "\uC2AC\uB85C\uBCA0\uB2C8\uC544", lat: 46.0569, lon: 14.5058 },
  { name: "\uBCA0\uC624\uADF8\uB77C\uB4DC", country: "\uC138\uB974\uBE44\uC544", lat: 44.7866, lon: 20.4489 },
  { name: "\uC0AC\uB77C\uC608\uBCF4", country: "\uBCF4\uC2A4\uB2C8\uC544", lat: 43.8563, lon: 18.4131 },
  { name: "\uD2F0\uB77C\uB098", country: "\uC54C\uBC14\uB2C8\uC544", lat: 41.3275, lon: 19.8187 },
  // ── 북유럽 ──
  { name: "\uD5EC\uC2F1\uD0A4", country: "\uD540\uB780\uB4DC", lat: 60.1699, lon: 24.9384 },
  { name: "\uC2A4\uD1A1\uD640\uB984", country: "\uC2A4\uC6E8\uB374", lat: 59.3293, lon: 18.0686 },
  { name: "\uC624\uC2AC\uB85C", country: "\uB178\uB974\uC6E8\uC774", lat: 59.9139, lon: 10.7522 },
  { name: "\uCF54\uD39C\uD558\uAC90", country: "\uB374\uB9C8\uD06C", lat: 55.6761, lon: 12.5683 },
  { name: "\uB808\uC774\uCEAC\uBE44\uD06C", country: "\uC544\uC774\uC2AC\uB780\uB4DC", lat: 64.1466, lon: -21.9426 },
  { name: "\uB354\uBE14\uB9B0", country: "\uC544\uC77C\uB79C\uB4DC", lat: 53.3498, lon: -6.2603 },
  // ── 발트 3국 ──
  { name: "\uD0C8\uB9B0", country: "\uC5D0\uC2A4\uD1A0\uB2C8\uC544", lat: 59.437, lon: 24.7536 },
  { name: "\uB9AC\uAC00", country: "\uB77C\uD2B8\uBE44\uC544", lat: 56.9496, lon: 24.1052 },
  { name: "\uBE4C\uB274\uC2A4", country: "\uB9AC\uD22C\uC544\uB2C8\uC544", lat: 54.6872, lon: 25.2797 },
  // ── 동유럽 / CIS ──
  { name: "\uBAA8\uC2A4\uD06C\uBC14", country: "\uB7EC\uC2DC\uC544", lat: 55.7558, lon: 37.6173 },
  { name: "\uC0C1\uD2B8\uD398\uD14C\uB974\uBD80\uB974\uD06C", country: "\uB7EC\uC2DC\uC544", lat: 59.9311, lon: 30.3609 },
  { name: "\uBE14\uB77C\uB514\uBCF4\uC2A4\uD1A0\uD06C", country: "\uB7EC\uC2DC\uC544", lat: 43.1332, lon: 131.9113 },
  { name: "\uD0A4\uC774\uC6B0", country: "\uC6B0\uD06C\uB77C\uC774\uB098", lat: 50.4504, lon: 30.5245 },
  { name: "\uBBFC\uC2A4\uD06C", country: "\uBCA8\uB77C\uB8E8\uC2A4", lat: 53.9006, lon: 27.559 },
  { name: "\uD0A4\uC2DC\uB108\uC6B0", country: "\uBAB0\uB3C4\uBC14", lat: 47.0105, lon: 28.8638 },
  // ── 북아메리카 ──
  { name: "\uC6CC\uC2F1\uD134 D.C.", country: "\uBBF8\uAD6D", lat: 38.9072, lon: -77.0369 },
  { name: "\uB274\uC695", country: "\uBBF8\uAD6D", lat: 40.7128, lon: -74.006 },
  { name: "\uB85C\uC2A4\uC564\uC824\uB808\uC2A4", country: "\uBBF8\uAD6D", lat: 34.0522, lon: -118.2437 },
  { name: "\uC2DC\uCE74\uACE0", country: "\uBBF8\uAD6D", lat: 41.8781, lon: -87.6298 },
  { name: "\uC0CC\uD504\uB780\uC2DC\uC2A4\uCF54", country: "\uBBF8\uAD6D", lat: 37.7749, lon: -122.4194 },
  { name: "\uC2DC\uC560\uD2C0", country: "\uBBF8\uAD6D", lat: 47.6062, lon: -122.3321 },
  { name: "\uBCF4\uC2A4\uD134", country: "\uBBF8\uAD6D", lat: 42.3601, lon: -71.0589 },
  { name: "\uD638\uB180\uB8F0\uB8E8", country: "\uBBF8\uAD6D", lat: 21.3069, lon: -157.8583 },
  { name: "\uB77C\uC2A4\uBCA0\uC774\uAC70\uC2A4", country: "\uBBF8\uAD6D", lat: 36.1699, lon: -115.1398 },
  { name: "\uB9C8\uC774\uC560\uBBF8", country: "\uBBF8\uAD6D", lat: 25.7617, lon: -80.1918 },
  { name: "\uD734\uC2A4\uD134", country: "\uBBF8\uAD6D", lat: 29.7604, lon: -95.3698 },
  { name: "\uB308\uB7EC\uC2A4", country: "\uBBF8\uAD6D", lat: 32.7767, lon: -96.797 },
  { name: "\uC560\uD2C0\uB79C\uD0C0", country: "\uBBF8\uAD6D", lat: 33.749, lon: -84.388 },
  { name: "\uB374\uBC84", country: "\uBBF8\uAD6D", lat: 39.7392, lon: -104.9903 },
  { name: "\uC624\uD0C0\uC640", country: "\uCE90\uB098\uB2E4", lat: 45.4215, lon: -75.6972 },
  { name: "\uD1A0\uB860\uD1A0", country: "\uCE90\uB098\uB2E4", lat: 43.6532, lon: -79.3832 },
  { name: "\uBC34\uCFE0\uBC84", country: "\uCE90\uB098\uB2E4", lat: 49.2827, lon: -123.1207 },
  { name: "\uBAAC\uD2B8\uB9AC\uC62C", country: "\uCE90\uB098\uB2E4", lat: 45.5017, lon: -73.5673 },
  { name: "\uBA55\uC2DC\uCF54\uC2DC\uD2F0", country: "\uBA55\uC2DC\uCF54", lat: 19.4326, lon: -99.1332 },
  { name: "\uCE78\uCFE4", country: "\uBA55\uC2DC\uCF54", lat: 21.1619, lon: -86.8515 },
  // ── 카리브해 / 중앙아메리카 ──
  { name: "\uC544\uBC14\uB098", country: "\uCFE0\uBC14", lat: 23.1136, lon: -82.3666 },
  { name: "\uC0B0\uD1A0\uB3C4\uBC0D\uACE0", country: "\uB3C4\uBBF8\uB2C8\uCE74\uACF5\uD654\uAD6D", lat: 18.4861, lon: -69.9312 },
  { name: "\uACFC\uD14C\uB9D0\uB77C\uC2DC\uD2F0", country: "\uACFC\uD14C\uB9D0\uB77C", lat: 14.6349, lon: -90.5069 },
  { name: "\uC0B0\uD638\uC138", country: "\uCF54\uC2A4\uD0C0\uB9AC\uCE74", lat: 9.9281, lon: -84.0907 },
  { name: "\uD30C\uB098\uB9C8\uC2DC\uD2F0", country: "\uD30C\uB098\uB9C8", lat: 8.9824, lon: -79.5199 },
  { name: "\uC0B0\uD6C4\uC548", country: "\uD478\uC5D0\uB974\uD1A0\uB9AC\uCF54", lat: 18.4655, lon: -66.1057 },
  { name: "\uD0B9\uC2A4\uD134", country: "\uC790\uBA54\uC774\uCE74", lat: 18.0179, lon: -76.8099 },
  // ── 남아메리카 ──
  { name: "\uBCF4\uACE0\uD0C0", country: "\uCF5C\uB86C\uBE44\uC544", lat: 4.711, lon: -74.0721 },
  { name: "\uB9AC\uB9C8", country: "\uD398\uB8E8", lat: -12.0464, lon: -77.0428 },
  { name: "\uD0A4\uD1A0", country: "\uC5D0\uCF70\uB3C4\uB974", lat: -0.1807, lon: -78.4678 },
  { name: "\uC0B0\uD2F0\uC544\uACE0", country: "\uCE60\uB808", lat: -33.4489, lon: -70.6693 },
  { name: "\uBD80\uC5D0\uB178\uC2A4\uC544\uC774\uB808\uC2A4", country: "\uC544\uB974\uD5E8\uD2F0\uB098", lat: -34.6037, lon: -58.3816 },
  { name: "\uBAAC\uD14C\uBE44\uB370\uC624", country: "\uC6B0\uB8E8\uACFC\uC774", lat: -34.9011, lon: -56.1645 },
  { name: "\uC544\uC21C\uC2DC\uC628", country: "\uD30C\uB77C\uACFC\uC774", lat: -25.2637, lon: -57.5759 },
  { name: "\uB77C\uD30C\uC2A4", country: "\uBCFC\uB9AC\uBE44\uC544", lat: -16.4897, lon: -68.1193 },
  { name: "\uBE0C\uB77C\uC9C8\uB9AC\uC544", country: "\uBE0C\uB77C\uC9C8", lat: -15.7975, lon: -47.8919 },
  { name: "\uC0C1\uD30C\uC6B8\uB8E8", country: "\uBE0C\uB77C\uC9C8", lat: -23.5505, lon: -46.6333 },
  { name: "\uB9AC\uC6B0\uB370\uC790\uB124\uC774\uB8E8", country: "\uBE0C\uB77C\uC9C8", lat: -22.9068, lon: -43.1729 },
  { name: "\uCE74\uB77C\uCE74\uC2A4", country: "\uBCA0\uB124\uC218\uC5D8\uB77C", lat: 10.4806, lon: -66.9036 },
  // ── 아프리카 ──
  { name: "\uCE74\uC774\uB85C", country: "\uC774\uC9D1\uD2B8", lat: 30.0444, lon: 31.2357 },
  { name: "\uB098\uC774\uB85C\uBE44", country: "\uCF00\uB0D0", lat: -1.2921, lon: 36.8219 },
  { name: "\uC544\uB514\uC2A4\uC544\uBC14\uBC14", country: "\uC5D0\uD2F0\uC624\uD53C\uC544", lat: 9.025, lon: 38.7469 },
  { name: "\uC544\uD06C\uB77C", country: "\uAC00\uB098", lat: 5.6037, lon: -0.187 },
  { name: "\uB77C\uACE0\uC2A4", country: "\uB098\uC774\uC9C0\uB9AC\uC544", lat: 6.5244, lon: 3.3792 },
  { name: "\uC544\uBD80\uC790", country: "\uB098\uC774\uC9C0\uB9AC\uC544", lat: 9.0579, lon: 7.4951 },
  { name: "\uB2E4\uB974\uC5D0\uC2A4\uC0B4\uB78C", country: "\uD0C4\uC790\uB2C8\uC544", lat: -6.7924, lon: 39.2083 },
  { name: "\uCF00\uC774\uD504\uD0C0\uC6B4", country: "\uB0A8\uC544\uD504\uB9AC\uCE74\uACF5\uD654\uAD6D", lat: -33.9249, lon: 18.4241 },
  { name: "\uC694\uD558\uB124\uC2A4\uBC84\uADF8", country: "\uB0A8\uC544\uD504\uB9AC\uCE74\uACF5\uD654\uAD6D", lat: -26.2041, lon: 28.0473 },
  { name: "\uC54C\uC81C", country: "\uC54C\uC81C\uB9AC", lat: 36.7538, lon: 3.0588 },
  { name: "\uD280\uB2C8\uC2A4", country: "\uD280\uB2C8\uC9C0", lat: 36.8065, lon: 10.1815 },
  { name: "\uB77C\uBC14\uD2B8", country: "\uBAA8\uB85C\uCF54", lat: 34.0209, lon: -6.8416 },
  { name: "\uCE74\uC0AC\uBE14\uB791\uCE74", country: "\uBAA8\uB85C\uCF54", lat: 33.5731, lon: -7.5898 },
  { name: "\uCE84\uD314\uB77C", country: "\uC6B0\uAC04\uB2E4", lat: 0.3476, lon: 32.5825 },
  { name: "\uB8E8\uC0AC\uCE74", country: "\uC7A0\uBE44\uC544", lat: -15.3875, lon: 28.3228 },
  { name: "\uD558\uB77C\uB808", country: "\uC9D0\uBC14\uBE0C\uC6E8", lat: -17.8252, lon: 31.0335 },
  { name: "\uB9C8\uD478\uD22C", country: "\uBAA8\uC7A0\uBE44\uD06C", lat: -25.9692, lon: 32.5732 },
  { name: "\uC548\uD0C0\uB098\uB098\uB9AC\uBCF4", country: "\uB9C8\uB2E4\uAC00\uC2A4\uCE74\uB974", lat: -18.8792, lon: 47.5079 },
  // ── 오세아니아 ──
  { name: "\uCE94\uBC84\uB77C", country: "\uD638\uC8FC", lat: -35.2809, lon: 149.13 },
  { name: "\uC2DC\uB4DC\uB2C8", country: "\uD638\uC8FC", lat: -33.8688, lon: 151.2093 },
  { name: "\uBA5C\uBC84\uB978", country: "\uD638\uC8FC", lat: -37.8136, lon: 144.9631 },
  { name: "\uBE0C\uB9AC\uC988\uBC88", country: "\uD638\uC8FC", lat: -27.4698, lon: 153.0251 },
  { name: "\uD37C\uC2A4", country: "\uD638\uC8FC", lat: -31.9505, lon: 115.8605 },
  { name: "\uC6F0\uB9C1\uD134", country: "\uB274\uC9C8\uB79C\uB4DC", lat: -41.2866, lon: 174.7756 },
  { name: "\uC624\uD074\uB79C\uB4DC", country: "\uB274\uC9C8\uB79C\uB4DC", lat: -36.8485, lon: 174.7633 }
];
var SEOUL = KOREAN_CITIES[0];
var AMBIGUOUS_NAMES = new Set(
  KOREAN_CITIES.map((c) => c.name).filter((name, _, arr) => arr.indexOf(name) !== arr.lastIndexOf(name))
);
function formatCityName(city) {
  if (city.country) return `${city.name}, ${city.country}`;
  if (city.region && AMBIGUOUS_NAMES.has(city.name)) return `${city.name} (${city.region})`;
  return city.name;
}
function filterCities(query) {
  const q = query.trim();
  if (!q) return [];
  const chosung = isAllChosung(q);
  const koreanResults = [];
  const worldResults = [];
  for (const city of KOREAN_CITIES) {
    if (chosung ? matchChosung(city.name, q) || city.region != null && matchChosung(city.region, q) : city.name.includes(q) || city.region != null && city.region.includes(q)) {
      koreanResults.push(city);
    }
  }
  for (const city of WORLD_CITIES) {
    if (chosung) {
      if (matchChosung(city.name, q)) worldResults.push(city);
    } else {
      const label = formatCityName(city);
      if (label.includes(q)) worldResults.push(city);
    }
  }
  return [...koreanResults, ...worldResults].slice(0, 20);
}

export {
  KOREAN_CITIES,
  WORLD_CITIES,
  SEOUL,
  formatCityName,
  filterCities
};
