// @orrery/core/dist/chunk-6PLRETU5.js
import {
  adjustKdtToKst
} from "./chunk-TW4ADZCX.js";
import {
  analyzeAllRelations,
  calculateInjongbeop,
  calculateJwabeop,
  getDaewoon,
  getFourPillars,
  getGongmang,
  getHiddenStems,
  getJeonggi,
  getRelation,
  getSpecialSals,
  getTwelveMeteor,
  getTwelveSpirit
} from "./chunk-JLKMDX3R.js";

// src/saju.ts
function getSipsin(dayStem, targetStem) {
  const rel = getRelation(dayStem, targetStem);
  return rel ? rel.hanja : "?";
}
function calculateSaju(input) {
  const kst = adjustKdtToKst(input.year, input.month, input.day, input.hour, input.minute);
  const { year, month, day, hour, minute } = kst;
  const { gender } = input;
  const isMale = gender === "M";
  const [yp, mp, dp, hp] = getFourPillars(year, month, day, hour, minute, input.jasiMethod);
  const dayStem = dp[0];
  const stems = [hp[0], dp[0], mp[0], yp[0]];
  const branches = [hp[1], dp[1], mp[1], yp[1]];
  const ganzis = [hp, dp, mp, yp];
  const pillars = ganzis.map((ganzi, i) => {
    const stem = stems[i];
    const branch = branches[i];
    let stemSipsin = getSipsin(dayStem, stem);
    if (i === 1) stemSipsin = "\u672C\u5143";
    const jeonggi = getJeonggi(branch);
    const branchSipsin = getSipsin(dayStem, jeonggi);
    const unseong = getTwelveMeteor(dayStem, branch);
    const sinsal = getTwelveSpirit(yp[1], branch);
    const jigang = getHiddenStems(branch);
    const pillar = { ganzi, stem, branch };
    return {
      pillar,
      stemSipsin,
      branchSipsin,
      unseong,
      sinsal,
      jigang
    };
  });
  const dwHour = input.unknownTime ? 12 : hour;
  const dwMinute = input.unknownTime ? 0 : minute;
  const rawDaewoon = getDaewoon(isMale, year, month, day, dwHour, dwMinute, input.jasiMethod);
  const yearBranch = yp[1];
  const gmBranches = getGongmang(dp);
  const gmSet = new Set(gmBranches);
  const gongmang = {
    branches: gmBranches,
    pillarIndices: branches.reduce((acc, b, i) => {
      if (i !== 1 && gmSet.has(b)) acc.push(i);
      return acc;
    }, [])
  };
  const daewoon = rawDaewoon.map((dw, i) => {
    const age = dw.startDate.getFullYear() - year;
    const dwStem = dw.ganzi[0];
    const dwBranch = dw.ganzi[1];
    const dwStemSipsin = getSipsin(dayStem, dwStem);
    const dwBranchJeonggi = getJeonggi(dwBranch);
    const dwBranchSipsin = getSipsin(dayStem, dwBranchJeonggi);
    const unseong = getTwelveMeteor(dayStem, dwBranch);
    const sinsal = getTwelveSpirit(yearBranch, dwBranch);
    return {
      index: i + 1,
      ganzi: dw.ganzi,
      startDate: dw.startDate,
      age,
      stemSipsin: dwStemSipsin,
      branchSipsin: dwBranchSipsin,
      unseong,
      sinsal,
      isGongmang: gmSet.has(dwBranch)
    };
  });
  const relations = analyzeAllRelations(ganzis);
  const specialSals = getSpecialSals(stems, branches, dp);
  const dayBranch = dp[1];
  const jwabeop = calculateJwabeop(dayStem, branches, dayBranch);
  const injongbeop = calculateInjongbeop(dayStem, dayBranch);
  return {
    input,
    pillars,
    daewoon,
    relations,
    specialSals,
    gongmang,
    jwabeop,
    injongbeop
  };
}

export {
  calculateSaju
};
