// @orrery/core/dist/constants.js
import {
  BAEKHO_PILLARS,
  BRANCH_BREAKS,
  BRANCH_CLASHES,
  BRANCH_COMBINES_6,
  BRANCH_ELEMENT,
  BRANCH_GWIMUN,
  BRANCH_HARMS,
  BRANCH_PUNISHMENTS,
  BRANCH_SELF_PUNISHMENTS,
  BRANCH_WONJIN,
  BRIGHTNESS_TABLE,
  CHANGSHENG_START,
  CHEONDUK_MAP,
  CHEONUL_MAP,
  DIRECTIONAL_COMPOSES,
  DIRECTIONAL_COMPOSE_ELEMENTS,
  DI_ZHI,
  DOHWA_MAP,
  EARTH,
  EARTH_KR,
  ELEMENT_HANJA,
  GEUMYEO_MAP,
  GOEGANG_PILLARS,
  GONGMANG_TABLE,
  HALF_COMPOSES,
  HGANJI,
  HONGYEOM_PILLARS,
  HUO_XING_START,
  JIJANGGAN,
  KUI_YUE_TABLE,
  LING_XING_START,
  LUCKY_STAR_NAMES,
  LU_CUN_TABLE,
  MAIN_STAR_NAMES,
  METEORS_12,
  METEOR_LOOKUP,
  MUNCHANG_MAP,
  NAYIN_TABLE,
  PALACE_NAMES,
  PILLAR_NAMES,
  RELATIONS,
  SHA_STAR_NAMES,
  SI_HUA_TABLE,
  SKY,
  SKY_KR,
  SPIRITS_12,
  STEM_CLASHES,
  STEM_COMBINES,
  STEM_INFO,
  TIANFU_SERIES_OFFSETS,
  TIAN_GAN,
  TIAN_MA_TABLE,
  TRIPLE_COMPOSES,
  TRIPLE_COMPOSE_ELEMENTS,
  WOLDUK_MAP,
  WU_HU_DUN_GAN,
  WU_XING_JU_MAP,
  YANGGAN,
  YANGIN_MAP,
  ZIWEI_SERIES_OFFSETS
} from "./chunk-EXYPMSUR.js";
import "./chunk-JSBRDJBE.js";
export {
  BAEKHO_PILLARS,
  BRANCH_BREAKS,
  BRANCH_CLASHES,
  BRANCH_COMBINES_6,
  BRANCH_ELEMENT,
  BRANCH_GWIMUN,
  BRANCH_HARMS,
  BRANCH_PUNISHMENTS,
  BRANCH_SELF_PUNISHMENTS,
  BRANCH_WONJIN,
  BRIGHTNESS_TABLE,
  CHANGSHENG_START,
  CHEONDUK_MAP,
  CHEONUL_MAP,
  DIRECTIONAL_COMPOSES,
  DIRECTIONAL_COMPOSE_ELEMENTS,
  DI_ZHI,
  DOHWA_MAP,
  EARTH,
  EARTH_KR,
  ELEMENT_HANJA,
  GEUMYEO_MAP,
  GOEGANG_PILLARS,
  GONGMANG_TABLE,
  HALF_COMPOSES,
  HGANJI,
  HONGYEOM_PILLARS,
  HUO_XING_START,
  JIJANGGAN,
  KUI_YUE_TABLE,
  LING_XING_START,
  LUCKY_STAR_NAMES,
  LU_CUN_TABLE,
  MAIN_STAR_NAMES,
  METEORS_12,
  METEOR_LOOKUP,
  MUNCHANG_MAP,
  NAYIN_TABLE,
  PALACE_NAMES,
  PILLAR_NAMES,
  RELATIONS,
  SHA_STAR_NAMES,
  SI_HUA_TABLE,
  SKY,
  SKY_KR,
  SPIRITS_12,
  STEM_CLASHES,
  STEM_COMBINES,
  STEM_INFO,
  TIANFU_SERIES_OFFSETS,
  TIAN_GAN,
  TIAN_MA_TABLE,
  TRIPLE_COMPOSES,
  TRIPLE_COMPOSE_ELEMENTS,
  WOLDUK_MAP,
  WU_HU_DUN_GAN,
  WU_XING_JU_MAP,
  YANGGAN,
  YANGIN_MAP,
  ZIWEI_SERIES_OFFSETS
};
